File: Duels/Models/Player.cs
namespace Duels.Models;

public class Player : IHasCreator
{

  public string Faction { get; set; }
  public int PlayerNum { get; set; }
  public int GameId { get; set; }
  public Game Game { get; set; }
  public string Status { get; set; }

  // IHasCreator Variables
  public int Id { get; set; }
  public string CreatorId { get; set; }
  public Profile Creator { get; set; }
  public DateTime CreatedAt { get; set; }
  public DateTime UpdatedAt { get; set; }
}

File: Duels.client/src/models/Player.js


export class Player {
  constructor(data) {
    // IHasCreator Variables
    this.id = data.id
    this.creatorId = data.creatorId
    this.creator = data.creator
    this.createdAt = data.createdAt
    this.updatedAt = data.updatedAt

    this.faction = data.faction
    this.playerNum = data.playerNum
    this.gameId = data.gameId
    this.game = data.game
    this.status = data.status
  }
}